test/functional/ui/incsubstitution_spec.lua

```diff
@@ -0,0 +1,379 @@
+-- Test the good behavior of the live action : substitution
+
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear, feed, insert = helpers.clear, helpers.feed, helpers.insert
+local execute = helpers.execute
+
+describe('Substitution', function()
+  local screen
+
+  before_each(function()
+    clear()
+    execute("syntax on")
+    execute("set incsubstitute=2")
+    execute('set nohlsearch')
+    execute("hi IncSubstitute guifg=red guibg=yellow")
+    screen = Screen.new(40, 20)  -- 40 lines of 40 char
+    screen:attach()
+    screen:set_default_attr_ignore( {{bold=true, foreground=Screen.colors.Blue}} )
+    screen:set_default_attr_ids({
+      [1]  = {foreground = Screen.colors.Fuchsia},
+      [2]  = {foreground = Screen.colors.Brown, bold = true},
+      [3]  = {foreground = Screen.colors.SlateBlue},
+      [4]  = {bold = true, foreground = Screen.colors.SlateBlue},
+      [5]  = {foreground = Screen.colors.DarkCyan},
+      [6]  = {bold = true},
+      [7]  = {underline = true, bold = true, foreground = Screen.colors.SlateBlue},
+      [8]  = {foreground = Screen.colors.Slateblue, underline = true},
+      [9]  = {background = Screen.colors.Yellow},
+      [10] = {reverse = true},
+      [11] = {reverse = true, bold=true},
+      [12] = {foreground = Screen.colors.Red, background = Screen.colors.Yellow}
+    })
+  end)
+
+  after_each(function()
+    screen:detach()
+  end)
+
+  -- ----------------------------------------------------------------------
+  -- simple tests
+  -- ----------------------------------------------------------------------
+
+  it('old behavior if :set incsubstitute=0', function()
```

Comment:
I don't think we should enforce that strictly. The BDD style can result in very obtuse wording.